src/main/java/uk/gov/ons/census/action/messaging/FulfilmentRequestReceiver.java

```diff
@@ -41,31 +46,50 @@ public FulfilmentRequestReceiver(
   @Transactional
   @ServiceActivator(inputChannel = "actionFulfilmentInputChannel")
   public void receiveEvent(ResponseManagementEvent event) {
-    Case fulfilmentCase = fetchFulfilmentCase(event);
     String fulfilmentCode = event.getPayload().getFulfilmentRequest().getFulfilmentCode();
 
     Optional<ActionType> actionType = determineActionType(fulfilmentCode);
     if (actionType.isEmpty()) {
       return;
     }
 
+    UUID caseId = event.getPayload().getFulfilmentRequest().getCaseId();
+
+    if (individualResponseRequestCodes.contains(fulfilmentCode)) {
+      caseId = UUID.fromString(event.getPayload().getFulfilmentRequest().getIndividualCaseId());
+
+      // Here be dragons
+
+      try {
+        /*
+        Case processor and action scheduler ingest this message at the same time.
+        While case processor is creating the case we expect this to throw an exception
+        as the case will probably not exist the first time the case is looked for.
+        The sleep will reduce excessive logging.
+         */
+        Thread.sleep(1000);
```

Comment:
While this is well commented and we had a lengthy discussion about it, this feels really yucky. Would it really be so much more effort to send a new message from Case Processor to Action Scheduler, containing everything that's needed?

Reply:
I've never been a fan of the whole heath robinson eventual consistency thing at all, I believe the whole calling APIs to create UACs is purely to satisfy quickly pumping out SMS.   I reckon it could all be nicer, though we did also discuss it at length months ago.  I'm happy to leave it like this for now, and when we either do nightly jobs or something else change/fix it.

Reply:
eventual consistency is fine. This is something different